django/core/handlers/asgi.py

```diff
@@ -19,6 +20,34 @@
 logger = logging.getLogger('django.request')
 
 
+async def _sync_to_async_iterator(iterator):
+    """
+    Iterate over a sync generator in an async context.
+    """
+    q = asyncio.Queue()
+
+    def sync_iterator():
+        for item in iterator:
+            q.put_nowait(item)
+
+    task = asyncio.create_task(sync_to_async(sync_iterator)())
+    try:
+        while True:
+            next_item = asyncio.create_task(q.get())
```

Comment:
It seems like you should be able to do this without creating a new task for each call to `get()` and instead awaiting `get()` directly. For example, you could add a sentinel value to the queue when the sync iterator has been exhausted and check whether the retrieved value is the sentinel to know whether any items are left.

Reply:
Sentinel value is a good idea! I think it will make the logic simpler. I'll work on making this change and post a comment when I've updated the PR.